### ISSUE TITLE
Ephesus: Channel account business logic

### ISSUE THREAD

**Issue by dobertRowneySr** [Joystream/joystream#3713]
Difference in between channel account logic in `ephesus` and `audit3` ?

Relates to:

* #3657 
* #3648

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202078489349362/1202229419052519) by [Unito](https://www.unito.io)

**Comment by dobertRowneySr** [comment]
Reward account logic for claiming channel payouts

| Ephesus | Audit3 |
| ----------- | ----------- |
| reward account for channel is used | internal (sub)module account is used |

Reward account logic for NFT related proceeds (when NFT owner is a channel) and royalties
| Ephesus | Audit3 |
| ----------- | ----------- |
| reward account for channel is used (using the rules established in #3650 | internal (sub)module account is used |

TL;DR in Audit3 `channel::reward_account` is dropped in favour of sub module account, in `ephesus` old reward account logic is still used

**Comment by Lezek123** [comment]
We should probably update this before https://github.com/Joystream/joystream/pull/3771, revenue splits doesn't really make sense if the channel owner has full access to channel reward account.

**Comment by Lezek123** [comment]
Another reason to update is that there's a critical bug currently in Ephesus allowing the channel owner to steal funds from any address by using the `withdraw_from_channel_balance` extrinsic (which also really only makes sense when channel reward account is a module subaccount, not just any arbitrary account specified by the owner).

**Comment by bedeho** [comment]
Thanks for this summary @ignazio-bovo 

> We should probably update this before https://github.com/Joystream/joystream/pull/3771, revenue splits doesn't really make sense if the channel owner has full access to channel reward account.

I agree, but how do we update this in `ephesus` without merging in `audit3` (which has tons of other stuff)? I hope that won't get too messy to reconcile later.

@ignazio-bovo can you make an issue for fixing this and start on it next?

 > Another reason to update is that there's a critical bug currently in Ephesus allowing the channel owner to steal funds from any address by using 
 
Is there an issue for this?

**Comment by Lezek123** [comment]
> Is there an issue for this?

There's no separate issue, I think the submodule accounts change was really supposed to be on `ephesus` in the first place, but it actually only made it to `audit3`.

It was supposed to be part of https://github.com/Joystream/joystream/issues/3263 (of which all the remaining points were addressed in https://github.com/Joystream/joystream/pull/3444 and already made it into `ephesus`)
